includes/class-micropub-endpoint.php

```diff
@@ -46,57 +42,66 @@ public static function get( $array, $key, $default = array() ) {
 		return $default;
 	}
 
-	/**
-	 * Adds some query vars
-	 *
-	 * @param array $vars
-	 * @return array
-	 */
-	public static function query_var( $vars ) {
-		$vars[] = 'micropub';
-		return $vars;
+	public static function register_route() {
+		$cls = get_called_class();
+		register_rest_route(
+			MICROPUB_NAMESPACE, '/endpoint', array(
+				array(
+					'methods'  => WP_REST_Server::CREATABLE,
+					'callback' => array( $cls, 'post_handler' ),
+				),
+				array(
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => array( $cls, 'query_handler' ),
+				),
+			)
+		);
 	}
 
 	/**
 	 * Parse the micropub request and render the document
 	 *
-	 * @param WP $wp WordPress request context
+	 * @param WP_REST_Request $request WordPress request
 	 *
 	 * @uses apply_filter() Calls 'before_micropub' on the default request
 	 */
-	public static function parse_query( $wp ) {
-		if ( ! array_key_exists( 'micropub', $wp->query_vars ) ) {
-			return;
-		}
+	protected static function load_input( $request ) {
+		static::$scopes                 = apply_filters( 'indieauth_scopes', static::$scopes );
+		static::$micropub_auth_response = apply_filters( 'indieauth_response', static::$micropub_auth_response );
+		$content_type                   = $request->get_header( 'content-type' );
 
-		static::load_input();
+		if ( 'GET' === $request->get_method() ) {
+			static::$input = $request->get_query_params();
+		} elseif ( 'application/json' === $content_type ) {
+			static::$input = $request->get_json_params();
+		} elseif ( ! $content_type ||
+			'application/x-www-form-urlencoded' === $content_type ||
+			'multipart/form-data' === $content_type ) {
+			static::$input = array();
+			foreach ( $request->get_body_params() as $key => $val ) {
+				if ( 'action' === $key || 'url' === $key ) {
+					static::$input[ $key ] = $val;
+				} elseif ( 'h' === $key ) {
+					static::$input['type'] = array( 'h-' . $val );
+				} elseif ( 'access_token' === $key ) {
+					continue;
+				} else {
+					static::$input['properties']         = self::get( static::$input, 'properties' );
+					static::$input['properties'][ $key ] =
+					( is_array( $val ) && wp_is_numeric_array( $val ) )
```

Comment:
minor: for readability, either inline these two lines onto line 90 or indent them?